src/base_model.jl

```diff
@@ -102,9 +112,26 @@ function build_base_model!(ECModel::AbstractEC, optimizer; use_notations=false)
             <= P_N_us_overestimate[u, t])
     # Design of assets of the user
     @variable(model_user,
-        0 <= x_us[u=user_set, a=device_names(users_data[u])]
+        0 <= n_us[u=user_set, a=device_names(users_data[u])]
             <= field_component(users_data[u], a, "max_capacity"))
 
+    # Set integer capacity
+    for u in user_set
+        for a in device_names(users_data[u])
+            if (has_component(users_data[u], a, "modularity") && field_component(users_data[u], a, "modularity") == true)
+                set_integer(n_us[u,a])
+                # It should be always true that if the component has a modularity, then a nominal capacity should be available
+                set_upper_bound(n_us[u,a], field_component(users_data[u], a, "max_capacity")/field_component(users_data[u], a, "nom_capacity"))

```

Comment:
Probably good to add the lower bound too: static 0 I'd say

Reply:
This should be already considered. In the lines 114-116 we define n_us having lower bound 0 and upper bound max_capacity. Then if the asset allows for the modularity we change only the upper_bound to max_capacity/nom_capacity.

Reply:
Ok :)